tests/unit/test_charm.py

```diff
@@ -7,651 +7,488 @@
-
-        with self.assertRaises(
-            WordPressBlockedStatusException, msg="core reconciliation should fail"
-        ):
-            self.harness.charm._core_reconciliation()
-        self.assertIsInstance(
-            self.harness.model.unit.status,
-            ops.charm.model.BlockedStatus,
-            "unit should be in WaitingStatus",
-        )
-        self.assertIn(
-            "db relation",
-            self.harness.model.unit.status.message,
-            "unit should wait for database connection info",
-        )
-
-    def test_core_reconciliation(self):
-        """
-        arrange: after peer relation established and database configured
-        act: run core reconciliation
-        assert: core reconciliation should update config files to match current config and
-            application state
-        """
-        self._setup_replica_consensus()
-        db_config = {
-            "db_host": "config_db_host",
-            "db_name": "config_db_name",
-            "db_user": "config_db_user",
-            "db_password": "config_db_password",
+            "host": harness.charm.state.relation_db_host,
+            "database": harness.charm.state.relation_db_name,
+            "user": harness.charm.state.relation_db_user,
+            "password": harness.charm.state.relation_db_password,
         }
-        self.patch.database.prepare_database(
-            host=db_config["db_host"],
-            database=db_config["db_name"],
-            user=db_config["db_user"],
-            password=db_config["db_password"],
-        )
-        self.harness.update_config(db_config)
-
-        self.assertTrue(
-            self.patch.database.is_wordpress_installed(db_config["db_host"], db_config["db_name"]),
-            "wordpress should be installed after core reconciliation",
-        )
-
-        db_config.update({"db_host": "config_db_host_2"})
-        self.patch.database.prepare_database(
-            host=db_config["db_host"],
-            database=db_config["db_name"],
-            user=db_config["db_user"],
-            password=db_config["db_password"],
-        )
-        self.harness.update_config({"db_host": "config_db_host_2"})
-
-        self.assertTrue(
-            self.patch.database.is_wordpress_installed("config_db_host_2", db_config["db_name"]),
-            "wordpress should be installed after database config changed",
-        )
 
-    @staticmethod
-    def _gen_action_event_mock():
-        event_mock = unittest.mock.MagicMock()
-        event_mock.set_results = unittest.mock.MagicMock()
-        event_mock.fail = unittest.mock.MagicMock()
-        return event_mock
-
-    def test_get_initial_password_action_before_replica_consensus(self):
-        """
-        arrange: before peer relation established but after charm created
-        act: run get-initial-password action
-        assert: get-initial-password action should fail
+    harness.begin_with_initial_hooks()
+
+    assert {None} == set(
```

Comment:
This is a picky comment and this could have already been coded in this way (there could also be other instances of this in the code which might be worth checking), it can be helpful to consistently put the value being checked on the left and the expected value on the right of the equality in `assert`, in this case it is obvious, in others it might not always be clear which value the test is asserting to be right if the assert fails

Reply:
Fixed, thanks!